modules/core/src/main/java/org/locationtech/jts/geom/impl/CoordinateArraySequence.java

```diff
@@ -178,9 +256,9 @@ public double getOrdinate(int index, int ordinateIndex)
     switch (ordinateIndex) {
       case CoordinateSequence.X:  return coordinates[index].x;
       case CoordinateSequence.Y:  return coordinates[index].y;
-      case CoordinateSequence.Z:  return coordinates[index].z;
+      default:
+	      return coordinates[index].getOrdinate(ordinateIndex);
```

Comment:
Sanity check, does one get Double.NaN or an exception if they go out of bounds?

Reply:
found it was a little inconsistent. My understanding is exception if you are out of bounds, DnD if you are in bounds buy the value is undefined.